interfaces/ssh-peers/peers.py

```diff
@@ -0,0 +1,55 @@
+from charmhelpers.core import hookenv
+from charms.reactive import hook
+from charms.reactive import RelationBase
+from charms.reactive import scopes
+from charms.reactive import bus
+
+
+class SshPeers(RelationBase):
+    """Allow public ssh keys to be propagated among units.
+
+    When a new peer connects, a new secret SSH key should be created,
+    and public key should be propagated to the peer using
+    set_local_public_key()
+
+    The peer will then be notified about the public key and can add it
+    to its authorized_keys file.
+    """
+
+    scope = scopes.UNIT
+
+    class states(bus.StateList):
+        connected = bus.State('{relation_name}.connected')
+        local_public_key = bus.State('{relation_name}.local_public_key')
+        new_remote_public_key = bus.State(
+            '{relation_name}.new_remote_public_key')
+
+    @hook('{peers:ssh-peers}-relation-{joined}')
+    def joined(self):
+        self.set_state(self.states.connected)
+
+    @hook('{peers:ssh-peers}-relation-{changed,joined}')
+    def changed(self):
+        previous_remote_public_key = self.get_local('remote-public-ssh-key')
+        new_remote_public_key = self.get_remote('public-ssh-key')
+        if not new_remote_public_key:
```

Comment:
Agreed, but I wanted to have this one first, since it's smallest. I changed it so that I do a return in this first if statement, so the else can be removed.